File: Data Structures & Algorithms/binary-tree-diameter/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    diameterOfBinaryTree(root) {
        if(root == null) return 0
        let length = 0;
function dfs(node){
    if(!node) return 0
            let left = dfs(node.left)
            let right = dfs(node.right);
            length = Math.max(length, left + right)
            return 1 + Math.max(left, right)
}
      dfs(root)    

        return length
    }
}

File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> l1;
    public void preOrder(TreeNode root) {
        if(root == null) return;
        l1.add(root.val);
        if(root.left != null) preOrder(root.left);
        if(root.right != null) preOrder(root.right);
    }
    public List<Integer> preorderTraversal(TreeNode root) {
        l1 = new ArrayList<>();
        preOrder(root);
        return l1;
    }
}